ctapipe/io/eventfilereader.py

```diff
@@ -148,216 +148,207 @@ def check_file_compatibility(file_path):
         """
 
     @property
-    @abstractmethod
-    def num_events(self):
+    def metadata(self):
         """
-        Abstract property to be defined in child class.
-
-        Obtain the number of events from the file, store it inside
-        self._num_events, and return the value.
+        A dictionary containing the metadata of the file. This could include:
+        * is_simulation (bool indicating if the file contains simulated events)
+        * Telescope:Camera names (list if file contains multiple)
+        * Information in the file header
+        * Observation ID
 
         Returns
         -------
-        self._num_events : int
+        dict
         """
+        return self._metadata
 
     @property
-    @abstractmethod
-    def event_id_list(self):
+    def is_stream(self):
```

Comment:
I was hoping in a future PR someone could define a method to check if a string in being read, in which case `__getitem__` and `__len__` are disabled, because in that circumstance it is impossible for them to operate as expected.

Reply:
So .. erm .. if this feature is planned ot be introduced in the future only ... maybe ... then. Do you really want to include it into this PR?

Let me ask like this: Could you implement a test, which shows that this feature is working correctly?

Also, this class does not have `__getitem__` and `__len__`. So .. is this method correct here?